### PR TITLE
Add compileJNI stage on Continuous test

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java [hunk 1]
```diff
@@ -11,19 +11,24 @@
...
+import org.testng.SkipException;
 import org.testng.annotations.Test;
 
-/** The file is for testing PyTorch MKLDNN functionalities. */
+/** The package is specific for PyTorch integration test. */
 public class MkldnnTest {
 
     @Test
-    public void testMKLDNNForPyTorch() {
+    public void testMKLDNN() {
```
Comment (zachgk) [inline integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:28]: Is there a reason you want to move this? I think it makes more sense to leave the integration module as focusing on the engine requirements of the DJL spec itself while tests that focus on engine specific things can be in their separate folders

Comment (stu1130) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:28]: we found PR #212 ran into a situation where it uses old JNI to run the test and obviously failed. By moving this to integration test, we will rebuild the JNI first then run it on `Continuous build` task

**Review by frankfliu — APPROVED** [review]

File: integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java (hunk 1 above)
Comment (zachgk) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: Is there a way we can check for stale JNI from gradle? Then, we can choose whether to download or compile JNI dynamically during the gradle build

Comment (stu1130) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: There are a few options.
1. check files change in those C++ code and java "header"
2. just recompile the jni one way or the other

Comment (stu1130) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: In addition, one of the advantage of putting pytorch specific tests in integrations is we can just run integration test for verification when we do PyTorch version upgrade

Comment (stu1130) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: Discussed with Frank offline, the windows JNI build is pretty slow. We want to keep the test fast.

Comment (zachgk) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: I am thinking something closer to 1. This isn't the first time the JNI caching has caused some weird problems.

Here is what I am thinking. Can we move the c tests to an isolated folder like `/pytorch/pytorch-native/src/main/jni` so we can hash the folder contents with something like `tar cf - . | gsha1sum`. Then, we upload the hash when we built the jni and check during download to know whether we use the hashed version or should recompile.

Comment (lanking520) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: I would say no to abandon Windows JNI build. Historically there are cases JNI passed in Mac and failed on Windows. It took me 4 days to fix them on Windows

Comment (zachgk) [reply integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java:40]: For #212, we can merge it without passing the tests to unblock the PR

**Comment by lanking520** [comment]
From my perspective, I would still recommend to put PyTorch specific test to PyTorch alone. Our integration test are designed to fit for every Deep Engine. It should cover the core functionalities we maintained. Like TF putting the ModelZoo test just in TF repo, we can do similar to that on PyTorch.

**Comment by stu1130** [comment]
Ok, I can add compileJNI in each stage of Continous build first then we do the file change check in a separate PR.

**Review by frankfliu — APPROVED** [review]

**Review by zachgk — APPROVED** [review]
